src/init.rs

```diff
@@ -22,24 +25,29 @@ pub fn init(template: Option<String>) -> Result<(), String> {
             // We have a template file that doesn't exist
             return Err(format!("The given template file at '{}' does not exist or can't be read. Please make sure the file exists and you have the permissions necessary to read from it.", template.as_ref().unwrap()));
         } else {
-            // Create a new `bonnie.toml` file using the default
-            // TODO read the default from `~/.bonnie/template.toml` if it exists
-            output = fs::write(
-                "./bonnie.toml",
-                format!(
-                    "version=\"{version}\"
+            // get t
+            let template = match template::get_default() {
+                Ok(template) => Ok(template),
+                Err(template::Error::Other(err)) if err.kind() == io::ErrorKind::NotFound => {
+                    Ok(format!(
+                        "version=\"{version}\"
 
 [scripts]
 start = \"echo \\\"No start script yet!\\\"\"
-                ",
-                    version = BONNIE_VERSION
-                ),
-            );
+",
+                        version = BONNIE_VERSION
+                    ))
+                }
+                Err(err) => Err(err),
+            }
+            .map_err(|err| format!("{:#?}", err))?;
```

Comment:
Please use well-formed, informative string errors at the site of the error. Then propagate them up to `main`.